src/Analyzers/MSTest.Analyzers/AssertionArgsShouldBePassedInCorrectOrderAnalyzer.cs

```diff
@@ -57,6 +57,9 @@ public override void Initialize(AnalysisContext context)
         });
     }
 
+    private static bool IsConstant(IArgumentOperation argumentOperation)
+        => argumentOperation.Value.ConstantValue.HasValue;
```

Comment:
NOTE: I removed the check for OperationKind.Literal.
If we have a literal, then ConstantValue.HasValue should always be true. All literals are constants (but not the other way around). The tests are already covering the case where we have a constant which is not literal.